examples/mongodbatlas_federated_settings_identity_provider/azure/README.md

```diff
@@ -0,0 +1,233 @@
+
+
+```mermaid
+---
+title: "Simplified OIDC flow"
+---
+sequenceDiagram
+    autonumber
+    participant U as User/App
+    participant A as Atlas
+    participant Idp as Identity Provider (IdP)
+
+    Note right of A: OIDC is configured and set on the organization<br>issuer_uri=https://{some-idp}.com/
+    A->>Idp: issuer_uri/.well-known/openid-configuration
+    Idp->>A: {..., "jwks_uri": "jwks_uri"}
+    A->>Idp: jwks_uri
+    Idp->>A: {keys: [{"kid": kid}, ...]}
+
+    Note right of U: Obtain token
+    U->>Idp: get_token(authentication-mechanism)
+    Idp->>U: JWT token {"iss": "issuer_uri", sub: "user-identity", "kid": "key-used, ...}
+
+    Note right of A: A database user is registered with idp_id/sub as username
+    U->>A: connect(JWT token)
+    A->>A: use kid and keys to validate the token<br>lookup sub with registered users
+    A->>U: auth ok
+```
+
+- Example of `issuer_uri` (1):
+  - <https://token.actions.githubusercontent.com/>
+  - <https://gitlab.com/>
+  - <https://sts.windows.net/{tenant-id}/>
+- Example of authentication mechanism (5)
+  - username+password
+  - two factor authentication
+  - certificate
+- JWT Token Authentication(8):
+  - Can use different fields in the `JWT` to authenticate
+
+### OIDC Workload Azure
+
+```mermaid
+---
+title: "OIDC Workload Azure"
+---
+sequenceDiagram
+    autonumber
+    participant VM as VM<br>(Azure)
+    participant A as Atlas
+    participant Idp as IdP<br>(Azure)
+    
+    Note right of A: OIDC is configured and set on the organization<br>Atlas will keep an internal {idp_id}
+    A->>Idp: https://sts.windows.net/{tenant-id}/.well-known/openid-configuration
+    Idp->>A: {..., "jwks_uri": "jwks_uri"}
+    A->>Idp: jwks_uri
+    Idp->>A: {keys: [{"kid": kid}, ...]}
+
+    Note right of VM: A python script (pymongo.py) runs on boot<br>It reads the MongoURI from an env-var<br>The pymongo driver gets the JWT from the metadata endpoint<br>See description below
+    VM->>Idp: http://169.254.169.254/metadata/identity/oauth2/token
+    Idp->>VM: JWT token: {<br>"iss": "https://sts.windows.net/{tenant-id}", <br>"sub": "{vm_configured_identity}"<br>}
+
+    Note right of A: A database user exist on the organization<br>username: {idp_id}/{vm_configured_identity}
+    VM ->> A: connect(JWT token, MongoURI)
+    activate A
+    A->>A: validate token and lookup database user
+    A->>VM: auth ok
+    VM->>A: insert record into database
+    A->>VM: insert record response
+    VM->>A: close connection
+    deactivate A
+```
+
+- (5) We use terraform to configure a [`user_assigned_identity`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/user_assigned_identity) for the VM and use the [metadata endpoint](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http) to obtain the token
+  - See also [pymongo docs](https://pymongo.readthedocs.io/en/stable/examples/authentication.html#mongodb-oidc)
```

Comment:
consider linking to https://www.mongodb.com/docs/languages/python/pymongo-driver/current/security/enterprise-authentication/#mongodb-oidc instead. we're trying to bring all of the drivers docs to the official docs site

Reply:
updated in b7a6e0f16a8c698054d8f5448177676575f9fe39